File: Omkar's Pet Cat/d1.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int T;
    cin >> T;

    while (T--) {
        int N;
        cin >> N;

        vector<long long> A(N);
        long long mn = LLONG_MAX;

        for (auto &x : A) {
            cin >> x;
            mn = min(mn, x);
        }

        bool hasNonDiv = false;
        int cntMin = 0;

        for (auto x : A) {
            if (x % mn != 0) hasNonDiv = true;
            if (x == mn) cntMin++;
        }

        if (hasNonDiv || cntMin == 1)
            cout << "YES\n";
        else
            cout << "NO\n";
    }

    return 0;
}

File: Omkar's Pet Cat/d1.py
import sys

def solve():
    data = list(map(int, sys.stdin.read().split()))
    idx = 0
    T = data[idx]; idx += 1
    out = []

    for _ in range(T):
        N = data[idx]; idx += 1
        A = data[idx:idx+N]; idx += N

        mn = min(A)
        has_non_div = any(x % mn != 0 for x in A)
        cnt_min = A.count(mn)

        out.append("YES" if has_non_div or cnt_min == 1 else "NO")

    sys.stdout.write("\n".join(out))

if __name__ == "__main__":
    solve()